### PR TITLE
Macro serializer

### PR DESCRIPTION
**Pull request by nzpr**
## Overview

This PR introduces compile time macro expansion generating serialisers for types used in the code.
First commit is not related and just to remove InetSocketAddress, which has side application effect (dns resolution), with custom case class. 

Main commit is the second one.
```scala
// Define your class
final case class MyObject(a: Int, b: String)

// Import derivation macro
import sdk.serialize.auto.*

// Instance is derived automatically.
val x = implicitly[Serialize[Eval, MyObject]]
```

In case of missing implicit error if instance cannot be derived, make sure that all fields of the class that are primitives (Int, String, etc) or sum types (Option, List, etc) have instances of Serialize in the specification defined in sdk.codecs.SerializePrimitiveTypes and sdk.codecs.SerializeSumTypes, respectively.

### Please make sure that this PR:

- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).

### CODE REVIEW

File: sdk/src/main/scala/sdk/codecs/SerializeDerived.scala [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+   * from the serialized data. If sum type has not specified above, runtime error will be raised.
+   */
+  def split[A](ctx: SealedTrait[Serialize[F, *], A]): Serialize[F, A] =
+    new Serialize[F, A] {
+      override def write(x: A): PrimitiveWriter[F] => F[Unit] = (_: PrimitiveWriter[F]) =>
+        assert(assertion = false, s"Serializers for sum types have to be defined manually $x.").pure[F]
+
+      override def read: PrimitiveReader[F] => F[A] = (_: PrimitiveReader[F]) =>
+        assert(assertion = false, s"Serializers for sum types have to be defined manually.").asInstanceOf[A].pure[F]
+    }
```
[the remark this answers is not in the document]
Comment (nzpr) [reply sdk/src/main/scala/sdk/codecs/SerializeDerived.scala]: Actually it turned out that it works if just comment out derivation for sum tyopes <https://github.com/nzpr/gorki-node/pull/29/commits/bf458ab8da47e804d8b53766c68c5666f991565d>. So there is a compile time check for correct serialiser!

**Comment by tgrospic** [comment]
Very nice that we can use simple macro for serialization! :tada:

File: macros/src/main/scala/sdk/codec/SerializeEvalInstances.scala [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * // Instance is derived automatically.
+ * val x = implicitly[Serialize[Eval, MyObject]]
+ * }}}
+ *
+ * In case of missing implicit error if instance cannot be derived,
+ * make sure that all fields of the class that are primitives (Int, String, etc)
+ * or sum types (Option, List, etc) have instances of Serialize in the specification
+ * defined in [[sdk.codecs.SerializePrimitiveTypes]] and [[sdk.codecs.SerializeSumTypes]], respectively.
+ */
+object SerializeEvalInstances extends SerializeDerivation[Eval] with SerializePrimitiveTypes[Eval]
```
Comment (tgrospic) [inline macros/src/main/scala/sdk/codec/SerializeEvalInstances.scala:27]: Let's call this `SerializeAutoDerivation` or something similar.

Comment (nzpr) [reply macros/src/main/scala/sdk/codec/SerializeEvalInstances.scala:27]: e.g. in circe it is called `auto` object

How about making this object `sdk.serialize.auto` ?

**Pull request merged**